homeassistant/components/netatmo.py

```diff
@@ -50,7 +64,10 @@ def setup(hass, config):
             "Please check your settings for NatAtmo API.")
         return False
 
-    for component in 'camera', 'sensor':
+    components = [component for component in ('camera', 'sensor')
```

Comment:
This breaks the auto discovery of the platforms unless there is a config for it. We want the camera to be automatically set up if there is one. 

We should detect if there is a camera by querying the netatmo data instead of the user config.